src/geometry/rtree.hpp

```diff
@@ -36,68 +31,61 @@ namespace bgi = bg::index;
 using Cartesian = bg::cs::cartesian;
 using Geographic = bg::cs::geographic<bg::degree>;
 
-using string = std::basic_string<char, std::char_traits<char>, Allocator::Allocator<char>>;
-
 template <typename CoordSystem>
-struct RTree {
-  using point_type =
-      bgm::point<double, 2, CoordSystem>;  // TODO: GEOMETRY - dimension template param (2 or 3)
+class RTree {
+ public:
+  // TODO: GEOMETRY - dimension template param (2 or 3)
+  using point_type = bgm::point<double, 2, CoordSystem>;
   // bgm::polygon requires default constructible allocators, allocations must be tracked by hand.
   using poly_type = bgm::polygon<point_type, true, true, std::vector, std::vector,
                                  Allocator::StatefulAllocator, Allocator::StatefulAllocator>;
   using geom_type = std::variant<point_type, poly_type>;
 
   using rect_type = bgm::box<point_type>;
   using doc_type = std::pair<rect_type, t_docId>;
+  using doc_alloc = Allocator::TrackingAllocator<doc_type>;
   using rtree_type = bgi::rtree<doc_type, bgi::quadratic<16>, bgi::indexable<doc_type>,
-                                bgi::equal_to<doc_type>, Allocator::TrackingAllocator<doc_type>>;
+                                bgi::equal_to<doc_type>, doc_alloc>;
 
-  using LUT_value_type = std::pair<t_docId const, geom_type>;
-  using LUT_type =
-      boost::unordered_flat_map<t_docId, geom_type, std::hash<t_docId>, std::equal_to<t_docId>,
-                                Allocator::TrackingAllocator<LUT_value_type>>;
+  using lookup_type = std::pair<t_docId const, geom_type>;
+  using lookup_alloc = Allocator::TrackingAllocator<lookup_type>;
+  using LUT_type = boost::unordered_flat_map<t_docId, geom_type, std::hash<t_docId>,
+                                             std::equal_to<t_docId>, lookup_alloc>;
 
   using query_results = std::vector<doc_type, Allocator::TrackingAllocator<doc_type>>;
 
-  Allocator::TrackingAllocator<void> alloc_;
+ private:
+  mutable std::size_t allocated_;
   rtree_type rtree_;
   LUT_type docLookup_;
 
-  RTree() = delete;
-  explicit RTree(size_t& alloc_ref);
-
-  [[nodiscard]] auto lookup(t_docId id) const
-      -> std::optional<std::reference_wrapper<const geom_type>>;
-  [[nodiscard]] auto lookup(doc_type const& doc) const
-      -> std::optional<std::reference_wrapper<const geom_type>>;
+ public:
+  explicit RTree();
 
-  [[nodiscard]] auto from_wkt(std::string_view wkt) const -> geom_type;
-  void insert(geom_type const& geom, t_docId id);
   int insertWKT(std::string_view wkt, t_docId id, RedisModuleString** err_msg);
   bool remove(t_docId id);
+  [[nodiscard]] auto query(std::string_view wkt, QueryType query_type,
+                           RedisModuleString** err_msg) const -> IndexIterator*;
 
-  [[nodiscard]] static auto geometry_to_string(geom_type const& geom) -> string;
```

Comment:
Consider moving to another file for doc services or somthing like this

Reply:
Next PR.